gui/codeeditor.cpp

```diff
@@ -97,10 +101,40 @@ void Highlighter::setSymbols(const QStringList &symbols)
         HighlightingRule rule;
         rule.pattern = QRegularExpression("\\b" + sym + "\\b");
         rule.format = mSymbolFormat;
+        rule.ruleRole = RuleRole::Symbol;
         mHighlightingRulesWithSymbols.append(rule);
     }
 }
 
+void Highlighter::setStyle( const CodeEditorStyle &newStyle )
+{
+    mKeywordFormat.setForeground( newStyle.keywordColor );
+    mKeywordFormat.setFontWeight( newStyle.keywordWeight );
+    mClassFormat.setForeground( newStyle.classColor );
+    mClassFormat.setFontWeight( newStyle.classWeight );
+    mSingleLineCommentFormat.setForeground( newStyle.commentColor );
+    mSingleLineCommentFormat.setFontWeight( newStyle.commentWeight );
+    mMultiLineCommentFormat.setForeground( newStyle.commentColor );
+    mMultiLineCommentFormat.setFontWeight( newStyle.commentWeight );
+    mQuotationFormat.setForeground( newStyle.quoteColor );
+    mQuotationFormat.setFontWeight( newStyle.quoteWeight );
+    mSymbolFormat.setForeground( newStyle.symbolFGColor );
+    mSymbolFormat.setBackground( newStyle.symbolBGColor );
+    mSymbolFormat.setFontWeight( newStyle.symbolWeight );
+    for( QVector<HighlightingRule>::iterator ruleIT = mHighlightingRules.begin();
```

Comment:
I wonder if you can't use foreach here.

Reply:
Tried that. Qt foreach macro wants to use a const variable in each iteration. That becomes a problem since contents are being altered in the call to apply the style. There is the std::foreach or std::for(auto &: <container) but IIRC that is only available >=c++11.

Reply:
ok.. please try C++11 range for if you can. We use that throughout the code in cppcheck-core so you can't build the GUI unless the compiler support that anyway.

Reply:
That's going to change a couple of things in the .pro and CMakeList.txt files. Are you okay with that?

Reply:
ok.. let's keep it like this for now.. and then somebody can try to modernize for loops later in a separate PR.

Reply:
Already had change incorporated and doing a full build for testing. Nothing jumps out in the logging messages. Expect this to be incorporated in final push. Only change was in .pro file where one line for gcc was using c++0x.